pkg/utils/utils.go

```diff
@@ -29,9 +30,13 @@ const (
 // Ensures that the generated chain name is exactly
 // maxChainLength chars in length
 func FormatChainName(name string, id string) string {
-	chainBytes := sha512.Sum512([]byte(name + id))
-	chain := fmt.Sprintf("%s%x", chainPrefix, chainBytes)
-	return chain[:maxChainLength]
+	return FormatChainNameWithPrefix(name, id, "")
+}
+
+func FormatChainNameWithPrefix(name string, id string, prefix string) string {
+	length := maxChainLength - prefixLength - len(prefix)
+	chain, _ := hash.New(name+id, length)
```

Comment:
Discarding this error seems risky.  The caller could pass in a long `prefix` string which could cause `length` to become less than 1.  Better to either
1. return the error up to the caller, or
2. alternately, have the `hash.New` internally `panic`.  But "[library functions should avoid panic](https://golang.org/doc/effective_go.html#panic)" and I'd only suggest doing that if we were confident that it was always a programmer error and not a user-input or environment issue.

Reply:
You're still discarding this error, no?